src/api/plans.ts

```diff
@@ -96,19 +108,68 @@ router.delete("/plans/:planId", async (req, res) => {
   res.send();
 });
 
-router.get("/plans/:planId/courses", (req, res) => {
-  res.send({
-    courseName: "string",
-    courseNumber: 4056,
-    units: 4,
-    crosslisted: 53,
-    prerequisites: [
-      {
-        needed: 3,
-        "courses:": [12, 52, 236],
-      },
-    ],
-    offered: ["F", "W", "SP", "SU"],
-    description: "course description",
+router.get("/plans/:planId/courses", async (req, res) => {
+  const planCourses = await sql<PlanInfo[]>`
+    WITH recursive R AS(
+    SELECT program_courses.course_id as course FROM plans
+    JOIN programs on plans.program_id = programs.id
```

Comment:
I think this JOIN may be unnecessary? Because both `program_courses.program_id` and `plans.program_id` are foreign key relations to `programs.id`. Since they're the same IDs you can just do something like this:
```sql
SELECT program_courses.course_id as course FROM program_courses
JOIN plans ON program_courses.program_id = plans.program_id
WHERE plans.id = 11
```